File: 3_examples/java/src/main/java/no/systek/graphqlworkshop/resolvers/DishResolver.java
package no.systek.graphqlworkshop.resolvers;

import com.coxautodev.graphql.tools.GraphQLResolver;
import java.util.Collection;
import no.systek.graphqlworkshop.storage.DataSource;
import no.systek.graphqlworkshop.storage.Dish;
import no.systek.graphqlworkshop.storage.Ingredient;
import org.springframework.stereotype.Component;

@Component
public class DishResolver implements GraphQLResolver<Dish> {

  private final DataSource dataSource;

  public DishResolver(DataSource dataSource) {
    this.dataSource = dataSource;
  }

  public Collection<Ingredient> ingredients(Dish dish) {
    return dataSource.findIngredientsInDish(dish.getId());
  }
}

File: 3_examples/kotlin/src/main/kotlin/no/systek/graphqlworkshop/resolvers/DishResolver.kt
package no.systek.graphqlworkshop.resolvers

import com.coxautodev.graphql.tools.GraphQLResolver
import no.systek.graphqlworkshop.storage.DataSource
import no.systek.graphqlworkshop.storage.Dish
import org.springframework.stereotype.Component

@Component
class DishResolver(
        private val dataSource: DataSource
) : GraphQLResolver<Dish> {
    fun ingredients(dish: Dish) = dataSource.findIngredientsInDish(dish.id)
}